examples/webserver/grpc/src/main/java/io/helidon/examples/webserver/grpc/StringServiceHealthCheck.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.examples.webserver.grpc;
+
+import io.helidon.common.configurable.Resource;
+import io.helidon.common.tls.Tls;
+import io.helidon.config.Config;
+import io.helidon.health.HealthCheck;
+import io.helidon.health.HealthCheckResponse;
+import io.helidon.health.HealthCheckType;
+import io.helidon.webclient.api.WebClient;
+import io.helidon.webclient.grpc.GrpcClient;
+
+class StringServiceHealthCheck implements HealthCheck {
+
+    private final WebClient webClient;
+
+    StringServiceHealthCheck(Config config) {
+        Tls clientTls = Tls.builder()
+                .trust(trust -> trust
+                        .keystore(store -> store
+                                .passphrase("password")
+                                .trustStore(true)
+                                .keystore(Resource.create("client.p12"))))
+                .build();
+        int serverPort = config.get("port").asInt().get();
+        this.webClient = WebClient.builder()
+                .tls(clientTls)
+                .baseUri("https://localhost:" + serverPort)
+                .build();
+    }
+
+    @Override
+    public String name() {
+        return StringService.class.getSimpleName();
+    }
+
+    @Override
+    public HealthCheckType type() {
+        return HealthCheckType.READINESS;
+    }
+
+    /**
+     * Self-invocation to verify gRPC endpoint is available and ready.
+     *
+     * @return health check response
+     */
+    @Override
+    public HealthCheckResponse call() {
+        try {
+            GrpcClient grpcClient = webClient.client(GrpcClient.PROTOCOL);
+            StringServiceGrpc.StringServiceBlockingStub service =
+                    StringServiceGrpc.newBlockingStub(grpcClient.channel());
+            Strings.StringMessage res = service.upper(
+                    Strings.StringMessage.newBuilder().setText("hello").build());
```

Comment:
Generally it's good if health checks run quickly (although that's less important than it used to be in 4.x with blocking virtual threads). Conceivably it could take a while to send this request and either get the response to detect the failure. Plus sending a no-op request on each health retrieval places _some_ load on the remote system.

Is there a way that other parts of the example that actually send meaningful requests and receive responses could update an indicator, visible to the health check code, that tells whether the most recent request worked or not? Then the health check would only have to look at that indicator to decide what kind of health response to return, rather than send a request of its own. 

Clearly the health check value would be based on stale and potentially incorrect data if a failure causes the _next_ request to fail, but if failures are relatively rare (if that can even be known) then maybe overall that's an acceptable trade-off.

Reply:
Upon further review, just a simple indicator would probably not work because the health check reporting `down` might not permit further requests to be sent to the server, thereby preventing the server from sending a gRPC request and then clearing the indicator upon success.

At the risk of complicating what is supposed to be a simple example, maybe there is a good way to combine a simple use of fault tolerance around sending the gRPC request with the health check?